In [2]:
import numpy as np
from scipy import stats
from scipy.stats import norm

### Topics: Normal distribution, Functions of Random Variables

#### Q1.The time required for servicing transmissions is normally distributed with µ  = 45 minutes and σ = 8 minutes. The service manager plans to have work begin on the transmission of a customer’s car 10 minutes after the car is dropped off and the customer is told that the car will be ready within 1 hour from drop-off. What is the probability that the service manager cannot meet his commitment? 

Ans:-
#### µ  = 45 minutes
#### σ = 8 minutes
#### let X = amount required to complete the repair of the car.
#### p(completeing the work in 1 hr and starting the work after 10 min) = 60 - 10 = 50mins
#### ie,p(X<=50) -> p(X>50) -> 1-p(x<=50)


In [2]:
z = (50-45)/8
z

0.625

In [3]:
#probability that the service manager cannot meet his commitment is
1-stats.norm.cdf(0.625)

0.26598552904870054

#### Q2.	The current age (in years) of 400 clerical employees at an insurance claims processing center is normally distributed with mean µ = 38 and Standard deviation σ =6. For each statement below, please specify True/False. If false, briefly explain why.
#### A.	More employees at the processing center are older than 44 than between 38 and 44.
#### B.	A training program for employees under the age of 30 at the center would be expected to attract about 36 employees.

Ans:- µ = 38  σ =6 N=400



In [8]:
#employess older than 44 p(X>44)
1-stats.norm.cdf(44,38,6)

0.15865525393145707

In [9]:
#employess older than 38 and 44 p(38<X<44)
stats.norm.cdf(44,38,6) - stats.norm.cdf(38,38,6) 

0.3413447460685429

##### A is False.Because the probability of employees age between 38 and 44 is more.

In [4]:
#employees under 30  - p(X<30)
stats.norm.cdf(30,38,6)

0.09121121972586788

In [5]:
#no: of employees attending training program from 400, N=400 is N*p(X<30) 
N=400
N*stats.norm.cdf(30,38,6)

36.484487890347154

##### B is True.Employees under the age of 30 at the center would attract about 36 employees.

In [ ]:
#------------------------------------------------------------------------------------------------------------------------------#

#### Q3.If X1 ~ N(μ, σ2^2) and X2 ~ N(μ, σ2^2) are iid normal random variables, then what is the difference between 2 X1 and X1 + X2? Discuss both their distributions and parameters.  

Ans: -
##### if X ∼ N(µ1, σ1^2 ), and Y ∼ N(µ2, σ2^2 ) = two independent random variables
##### then X + Y ∼ N(µ1 + µ2, σ1^2 + σ2^2 ) , and X − Y ∼ N(µ1 − µ2, σ1^2 + σ2^2 ) .
##### Similarly 
##### if Z = aX + bY 
##### then Z ∼ N(aµ1 + bµ2, a^2σ1^2 + b^2σ2^2 ).
##### Therefore 
##### 2X1~ N(2 u,4 σ^2)  ,  X1+X2 ~ N(µ + µ, σ^2 + σ^2 ) ~ N(2 u, 2σ^2 )
##### 2X1-(X1+X2) = N( 4µ,6 σ^2)

In [ ]:
#------------------------------------------------------------------------------------------------------------------------------#

#### Q4.	Let X ~ N(100, 20^2) . Find two values, a and b, symmetric about the mean, such that the probability of the random variable taking a value between them is 0.99. 

Ans:-
##### p(a<x<b)=0.99,µ=100,σ=20
##### Probability outside the a and b area is 1-0.99 = 0.01
##### probability towards left = -0.01/2 = -0.005
##### probability towards right = 0.01/2 = 0.005

In [10]:
#z value of 0.005
stats.norm.ppf(0.005)

-2.575829303548901

##### Z = (x-100)/20 
##### x = 20z+100
##### a = -(20 * 2.576)+100 = 48.5
##### b = (20 * 2.576)+100 = 151.52
##### ->a = 48.5, b = 151.52 - Option D

In [ ]:
#------------------------------------------------------------------------------------------------------------------------------#

#### Q5. Consider a company that has two different divisions. The annual profits from the two divisions are independent and have distributions Profit1 ~ N(5, 3^2) and Profit2 ~ N(7, 4^2) respectively. Both the profits are in Million. Answer the following questions about the total profit of the company in Rupees. Assume that $1 = Rs. 45
#### A.	Specify a Rupee range (centered on the mean) such that it contains 95% probability for the annual profit of the company.
Ans:- 1+0.95 =1.95/2 =0.975,
      1-0.95 =0.05/2 =0.025



In [17]:
stats.norm.ppf(0.025,45*5,3)

219.12010804637984

In [19]:
stats.norm.ppf(0.975,45*5,3)

230.87989195362016

In [21]:
stats.norm.ppf(0.025,45*7,4)

307.1601440618398

In [20]:
stats.norm.ppf(0.975,45*7,4)

322.8398559381602

95% probability for the annual profit of the company is given by =
profit1 +profit2
[219.12+230.87], [307.16+322.83]
[4449.99,629.99]

So Profit of the company in Rupees(in Million) is between range [449.99,629.99]

In [26]:
#B.Specify the 5th percentile of profit (in Rupees) for the company
stats.norm.ppf(0.05,45*5,3)

220.0654391191456

In [27]:
stats.norm.ppf(0.05,45*7,4)

308.4205854921941

In [28]:
#5th percentile of profit = 
220.065+308.42

528.485

In [29]:
#C.	Which of the two divisions has a larger probability of making a loss in a given year?

#p(division1 making a loss) p(x<0)
stats.norm.cdf(0,5,3)


0.0477903522728147

In [30]:
#p(division2 making a loss) p(x<0)
stats.norm.cdf(0,7,4)

0.040059156863817086

Divsion1 has large probabilty of making loss